soletta/simple_type_c.h

```diff
@@ -0,0 +1,228 @@
+/*
+ * This file is part of the Soletta (TM) Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "sol-flow.h"
+#include "sol-macros.h"
+
+/**
+ * @file
+ * @brief These routines are used for Soletta flow's @c Simple @c C type
+ */
+
+/**
+ * @defgroup SimpleCType Flow's Simple C Type
+ * @ingroup Flow
+ *
+ * @brief Helper to create a flow node type from a simple C function.
+ *
+ * @b Simple @b C @b Type is a helper to ease development of custom nodes where the full
+ * power of a node type is not needed. Instead, a single function is used
+ * and given to the node, the node private data and the event.
+ *
+ * Each node will have a context (private) data of size declared to
+ * sol_flow_simple_c_type_new_full(). This is given as the last
+ * argument to the callback @a func as well as can be retrieved with
+ * sol_flow_node_get_private_data(). An example:
+ *
+ * @code
+ * static int
+ * mytype_func(struct sol_flow_node *node,
+ *             const struct sol_flow_simple_c_type_event *ev,
+ *             void *data)
+ * {
+ *    struct my_context *ctx = data;
+ *    if (ev->type == SOL_FLOW_SIMPLE_C_TYPE_EVENT_TYPE_OPEN) {
+ *       ctx->my_value = initial_value;
+ *       ctx->my_string = strdup("inital_string");
+ *    } else if (ev->type == SOL_FLOW_SIMPLE_C_TYPE_EVENT_TYPE_CLOSE) {
+ *       free(ctx->my_string);
+ *       // do not free(ctx), it's automatically deleted
+ *    } else {
+ *       printf("value=%d, string=%s\n", ctx->my_value, ctx->my_string);
```

Comment:
This is how the original file is, I just made minor initial changes to implement the port description through array argument.